mart/utils/modality_dispatch.py

```diff
@@ -0,0 +1,68 @@
+#
+# Copyright (C) 2022 Intel Corporation
+#
+# SPDX-License-Identifier: BSD-3-Clause
+#
+
+from __future__ import annotations
+
+from itertools import cycle
+from typing import Any, Callable, Iterable
+
+import torch
+
+__all__ = ["modality_dispatch"]
+
+DEFAULT_MODALITY = "default"
+
+
+def modality_dispatch(
+    modality_func: Callable | dict[str, Callable],
+    data: torch.Tensor | dict[str, torch.Tensor] | Iterable[Any],
+    *,
+    input: torch.Tensor | dict[str, torch.Tensor] | Iterable[Any],
+    target: torch.Tensor | Iterable[Any] | None,
```

Comment:
These types should match whatever is in `Enforcer` above.

Reply:
Huh? Whatever enforcer takes should be the same things this function takes? If you believe that, then why are the type annotations different?

Reply:
But so does Enforcer? I can pass a tensor, a iterable[tensor] or iterable[dict[str, tensor]] to Enforcer. I don't get why the function being recursive matters. The type annotations tell you what the function accepts.

Reply:
I made it the same as in Enforcer.

Reply:
Well, it is now different from Enforcer, because other modules such as Initializer also use `modality_dispatch()` and they may have different types of arguments.